### PR TITLE
[BI-1056] All Germplasm Table

### PR DESCRIPTION
**Pull request by ctucker3**
# Description
**Story:** (https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1056)

Includes: 
- All germplasm table
  - I used the brapi models from Tim's reporting prototype for the germplasm models. https://github.com/Breeding-Insight/bi-web/tree/feature/reporting
- Germplasm navigation
- Germplasm lists navigation was included and the component will just have to be added in when ready. 

# Dependencies
sgn: BI-1056 (https://github.com/Breeding-Insight/sgn/pull/50)
bi-api: BI-1056 (https://github.com/Breeding-Insight/bi-api/pull/130)
bi-web: BI-1056 (https://github.com/Breeding-Insight/bi-web/pull/165)

# Testing
Use the importer and the file below to import some germplasm. Play around with the table and try some paging out to make sure everything works. 

[BI-1056_lots_of_germplasm.xls](https://github.com/Breeding-Insight/bi-web/files/7822502/BI-1056_lots_of_germplasm.xls)

# TAF Results
The taf job seems to be broken and is not printing out results. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [x] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [x] I have run TAF.

### CODE REVIEW

File: src/breeding-insight/service/BrAPIService.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from "@/util/api";
+import {BiResponse, Response} from "@/breeding-insight/model/BiResponse";
+import {ResultGenerator} from "@/breeding-insight/model/Result";
+import {Germplasm} from "@/breeding-insight/brapps/reporter/model/germplasm";
```
Comment (timparsons) [inline src/breeding-insight/service/BrAPIService.ts:21]: These imports aren't used and could be removed

File: src/views/germplasm/Germplasm.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    <section>
+      <nav class="tabs is-boxed">
+        <ul>
+          <router-link
+              v-bind:to="{name: 'germplasm-all', params: {programId: activeProgram.id}}"
+              tag="li"
+              active-class="is-active"
+          >
+            <a>Germplasm</a>
```
Comment (timparsons) [inline src/views/germplasm/Germplasm.vue:32]: Per the story, this text should be "All Germplasm"

File: src/router/index.ts [hunk 3]
```diff
@@ -259,6 +261,37 @@ const routes = [
...
+    },
+    component: Germplasm,
+    redirect: {name: 'germplasm-all'},
+    beforeEnter: processProgramNavigation,
+    children: [
+      {
+        path: 'germplasm-all',
+        name: 'germplasm-all',
+        meta: {
+          title: 'Germplasm',
```
Comment (timparsons) [inline src/router/index.ts:279]: Per the techspecs, this text should be "All Germplasm"

File: src/views/germplasm/GermplasmTable.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
+<template>
+  <section id="germplasmTable">
+    <h1 class="title">
+      Germplasm
```
Comment (timparsons) [inline src/views/germplasm/GermplasmTable.vue:4]: This should probably be "All Germplasm" to match the nav

**Review by timparsons — CHANGES_REQUESTED** [review]
The BrAPI models that were added are in the directory "brapps/reporter/model".  It might be better to move them to "brapi/model".

File: src/breeding-insight/service/BrAPIService.ts (hunk 1 above)
Comment (ctucker3) [reply src/breeding-insight/service/BrAPIService.ts]: Removed

File: src/views/germplasm/Germplasm.vue (hunk 2 above)
Comment (ctucker3) [reply src/views/germplasm/Germplasm.vue]: Changed

File: src/router/index.ts (hunk 3 above)
Comment (ctucker3) [reply src/router/index.ts]: Changed

**Review by timparsons — APPROVED** [review]

**Review by davedrp — APPROVED** [review]

**Pull request merged**